src/compiler/binder.ts

```diff
@@ -1931,7 +1934,11 @@ namespace ts {
 
                 case SyntaxKind.SpreadElementExpression:
                 case SyntaxKind.JsxSpreadAttribute:
-                    emitFlags |= NodeFlags.HasSpreadAttribute;
+                    let root = container;
+                    while (root && root.kind !== SyntaxKind.BinaryExpression) {
+                        root = root.parent;
+                    }
+                    emitFlags |= root && isDestructuringAssignment(root) ? NodeFlags.HasRestAttribute : NodeFlags.HasSpreadAttribute;

```

Comment:
I improved it to check that the relation between root and parent is that the object literal is the lefthand side.